### ISSUE TITLE
Debugging angular-cli project is broken

### ISSUE THREAD

**Issue by sinapis** [Microsoft/vscode-chrome-debug#257]
I get the following errors:

> ******** Error in DebugAdapter - Unhandled promise rejection: TypeError: Cannot read property 'locations' of undefined
>     at _chromeConnection.debugger_setBreakpointByUrlRegex.then.response (C:\Users\Amir\.vscode\extensions\msjsdiag.debugger-for-chrome-1.1.0\node_modules\vscode-chrome-debug-core\out\src\chrome\chromeDebugAdapter.js:376:54)

When trying to clear all break points, I get the following error:

> TypeError: Cannot read property 'reduce' of null
>     at ChromeDebugAdapter.clearAllBreakpoints (C:\Users\Amir\.vscode\extensions\msjsdiag.debugger-for-chrome-1.1.0\node_modules\vscode-chrome-debug-core\out\src\chrome\chromeDebugAdapter.js:349:56)
>     at setBreakpointsPFailOnError._setBreakpointsRequestQ.then (C:\Users\Amir\.vscode\extensions\msjsdiag.debugger-for-chrome-1.1.0\node_modules\vscode-chrome-debug-core\out\src\chrome\chromeDebugAdapter.js:293:38)
>     at process._tickCallback (internal/process/next_tick.js:103:7)

And breakpoints are not respected by the debugger (they appear grey with the message: "unverified brakpoint")

OS: win32 ia32
Node: v6.5.0
vscode-chrome-debug-core: 2.0.2
debugger-for-chrome: 1.1.0

Repro steps:
ng new angular2-app
ng serv
debug with the following launch.json:
```
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Launch Chrome against localhost, with sourcemaps",
      "type": "chrome",
      "request": "launch",
      "url": "http://localhost:4200/",
      "sourceMaps": true,
      "webRoot": "${workspaceRoot}/src",
      "diagnosticLogging": true,
      "runtimeArgs": [
        "--disable-session-crashed-bubble",
        "--disable-infobars"
      ],
      "userDataDir": "c:\\temp\\chorme_out_dir",
      "sourceMapPathOverrides": {
         "webpack:///C:*":"C:/*"
      }
    }
  ]
}
```

**Comment by roblourens** [comment]
Hm, I'm not seeing that. I'm about to update with a fix that might be relevant, but I think this is different. In the meantime, can I see the full log?

**Comment by roblourens** [comment]
1.2.0, let me know if you see anything different

**Comment by sinapis** [comment]
One thing changes, I no longer see an exception when I remove breakpoints.

However, all the rest is the same. Update repro steps:

put breakpoint on line 11 of C:\Angular2-app\src\app\app.component.ts line
Launch debugger

Expected: Debugger will stop on BP line.
Actual: 
1. Debugger stops on line 596 on zones.js file, then on main.bundle.js line 42501 and that's it.
2. BP is moved to line from line 11 to line 17.

Now, if I set BP on line 14 of app.component.ts line I'll get:

> TypeError: Cannot read property 'locations' of undefined
>     at _chromeConnection.debugger_setBreakpointByUrlRegex.then.response (C:\Users\Amir\.vscode\extensions\msjsdiag.debugger-for-chrome-1.2.0\node_modules\vscode-chrome-debug-core\out\src\chrome\chromeDebugAdapter.js:377:54)

Project can be found here: 
[Angular2-app.zip](https://github.com/Microsoft/vscode-chrome-debug/files/528762/Angular2-app.zip)

Debug console output:

[vscode-chrome-debug.txt](https://github.com/Microsoft/vscode-chrome-debug/files/528772/vscode-chrome-debug.txt)

**Comment by roblourens** [comment]
Two problems, the first is that your breakpoints aren't going to the right place, which has got to be the same as #250, so I'll merge these two. Second is an issue with error handling which is fixed already but not released.

**Comment by sinapis** [comment]
I see you closed #250. Is there an issue I can track the unified problem?

**Comment by roblourens** [comment]
I'd meant to reopen it, thanks for pointing that out
